src/utils/getIndentString.ts

```diff
@@ -1,12 +1,38 @@
-import { Bundle as MagicStringBundle } from 'magic-string';
-
-export default function getIndentString(
-	magicString: MagicStringBundle,
-	options: { indent?: boolean }
-) {
-	if (options.indent === true) {
-		return magicString.getIndentString();
+import Module from '../Module';
+
+function guessIndentString(code: string) {
+	const lines = code.split('\n');
+
+	const tabbed = lines.filter(line => /^\t+/.test(line));
+	const spaced = lines.filter(line => /^ {2,}/.test(line));
+
+	if (tabbed.length === 0 && spaced.length === 0) {
+		return null;
+	}
+
+	// More lines tabbed than spaced? Assume tabs, and
+	// default to tabs in the case of a tie (or nothing
+	// to go on)
+	if (tabbed.length >= spaced.length) {
+		return '\t';
+	}
+
+	// Otherwise, we need to guess the multiple
+	const min = spaced.reduce((previous, current) => {
+		const numSpaces = /^ +/.exec(current)[0].length;
+		return Math.min(numSpaces, previous);
+	}, Infinity);
+
+	return new Array(min + 1).join(' ');
+}
+
+export default function getIndentString(modules: Module[], options: { indent?: boolean }) {
+	if (options.indent !== true) return options.indent || '';
+
+	for (let i = 0; i < modules.length; i++) {
+		const indent = guessIndentString(modules[i].originalCode);
+		if (indent !== null) return indent;
 	}
 
-	return options.indent || '';
+	return '\t';
```

Comment:
I must admit I did not have a look at the original magic-string code but checking *all* lines in each file seems a little like overkill for me especially when weighing the performance cost of this vs. just getting the "right" indentation. Instead, we might e.g. write a loop that uses `.indexOf` to check for line-breaks and for each new line, it checks if it begins with a space or a tab until we have e.g. five lines of one category (=the "winner") and then use the lowest number of tabs or spaces used within those lines as indent.

Reply:
I copied the code verbatim to avoid possible regressions, but definitely it could be worth refining this code in a separate PR.

I'd even suggest applying whatever changes are made back to the original code in MagicString too. Perhaps this could be split out as a utility method there and imported directly as a submodule or something similar.